### PR TITLE
allow searching by mutations

### PR DESCRIPTION
**Pull request by bprize15**
resolves https://github.com/oncokb/oncokb-pipeline/issues/465

### CODE REVIEW

File: src/main/webapp/app/components/tabs/CurationHistoryTab.tsx [hunk 1]
```diff
@@ -50,6 +51,32 @@ const CurationHistoryTab = observer(({ historyData, getUsers, users, historyTabS
...
+    for (const mutation of mutations || []) {
+      for (const alteration of mutation.alterations || []) {
+        alts.add(alteration.alteration);
+      }
+      mutation.name.split(',').forEach(alt => alts.add(alt.trim()));
+    }
+    return alts;
+  }, [mutations]);
+
+  const mutationOptions = useMemo(() => {
```
Comment (zhx828) [inline src/main/webapp/app/components/tabs/CurationHistoryTab.tsx:65]: Should this be sorted? Do we use any default sorting for Author Select?

**Review by zhx828 — CHANGES_REQUESTED** [review]
After selecting mutation and looking at history, I want to look at another mutation. But when I click on the select, the cursor is at the beginning of the previous search keyword I don't seem to be able to remove besides typing(I intuitively want to hit on the delete keystroke). Anyway we can make this more intuitive?

File: src/main/webapp/app/components/tabs/CurationHistoryTab.tsx (hunk 1 above)
Comment (bprize15) [reply src/main/webapp/app/components/tabs/CurationHistoryTab.tsx:65]: There is not default sorting for the author select. However, I have changed the author dropdown to be sorted alphabetically.

I decided to sort the mutations using the default sort we came up with. However, we can use any of our mutation sorts for this order, so let me know if you prefer another sort.

**Comment by bprize15** [comment]
> After selecting mutation and looking at history, I want to look at another mutation. But when I click on the select, the cursor is at the beginning of the previous search keyword I don't seem to be able to remove besides typing(I intuitively want to hit on the delete keystroke). Anyway we can make this more intuitive?



> After selecting mutation and looking at history, I want to look at another mutation. But when I click on the select, the cursor is at the beginning of the previous search keyword I don't seem to be able to remove besides typing(I intuitively want to hit on the delete keystroke). Anyway we can make this more intuitive?

I added some props do the `ReactSelect`. Now, backspace clears the select, and there is an `X` button to clear it as well.

File: src/main/webapp/app/components/tabs/CurationHistoryTab.tsx (hunk 1 above)
Comment (zhx828) [reply src/main/webapp/app/components/tabs/CurationHistoryTab.tsx:70]: The default sorting is a bit odd without the context within the collapsible. But we can leave it as it until someone complains.

**Review by zhx828 — APPROVED** [review]
LGTM thanks.

**Pull request merged**